pkg/console/operator/operator.go

```diff
@@ -176,38 +176,53 @@ func (c *consoleOperator) Sync(obj metav1.Object) error {
 		return err
 	}
 
-	// all configs needed to do a sync
 	if err := c.handleSync(operatorConfig, consoleConfig, infrastructureConfig); err != nil {
-		c.SyncStatus(c.ConditionFailing(operatorConfig, "SyncLoopError", "Operator sync loop failed to complete."))
 		return err
 	}
-	c.SyncStatus(c.ConditionNotFailing(operatorConfig))
+
 	return nil
 }
 
-func (c *consoleOperator) handleSync(originalOperatorConfig *operatorsv1.Console, consoleConfig *configv1.Console, infrastructureConfig *configv1.Infrastructure) error {
-	operatorConfig := originalOperatorConfig.DeepCopy()
+func (c *consoleOperator) handleSync(operatorConfig *operatorsv1.Console, consoleConfig *configv1.Console, infrastructureConfig *configv1.Infrastructure) error {
+	operatorConfigCopy := operatorConfig.DeepCopy()
 
-	switch operatorConfig.Spec.ManagementState {
+	switch operatorConfigCopy.Spec.ManagementState {
 	case operatorsv1.Managed:
 		logrus.Println("console is in a managed state.")
 		// handled below
 	case operatorsv1.Unmanaged:
 		logrus.Println("console is in an unmanaged state.")
-		c.SyncStatus(c.ConditionsManagementStateUnmanaged(operatorConfig))
+		c.SyncStatus(c.ConditionsManagementStateUnmanaged(operatorConfigCopy))
```

Comment:
Won't this hot-loop for unmanaged/removed state? I think we need an equals check like we do below for managed state.

Reply:
It doesn't hot loop (has been tested).  
I believe because `UpdateStatus` does a `client.Put()`, so the resource isn't changed if the value is the same thus no loop.

Reply:
I just tried, and it seems to hot loop:

![screencast 2019-04-23 09-59-36](https://user-images.githubusercontent.com/1167259/56587110-f58ec480-65ae-11e9-9715-e135ff5797a6.gif)

Reply:
It just did for me as well.
Puzzled on why it didn't before. I'll add the guard. Best be safe.

Reply:
> It just did for me as well.
> Puzzled on why it didn't before.

I don't think it's new :/